test/integration/proposal-checks/IntegrationTestBorrowSDAICollateral.sol

```diff
@@ -0,0 +1,461 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.13;
+
+import "@forge-std/Test.sol";
+
+import {LendingTerm} from "@src/loan/LendingTerm.sol";
+import {PostProposalCheckFixture} from "@test/integration/proposal-checks/PostProposalCheckFixture.sol";
+
+contract IntegrationTestBorrowSDAICollateral is PostProposalCheckFixture {
+    function testTermParamSetup() public {
+        assertEq(term.collateralToken(), address(sdai));
+        {
+            LendingTerm.LendingTermParams memory params = term.getParameters();
+            assertEq(params.collateralToken, address(sdai));
+            assertEq(params.openingFee, 0);
+            assertEq(params.interestRate, SDAI_RATE);
+            assertEq(params.minPartialRepayPercent, 0);
+            assertEq(params.maxDelayBetweenPartialRepay, 0);
+            assertEq(params.maxDebtPerCollateralToken, MAX_SDAI_CREDIT_RATIO);
+        }
+        {
+            LendingTerm.LendingTermReferences memory params = term
+                .getReferences();
+
+            assertEq(params.profitManager, addresses.mainnet("PROFIT_MANAGER"));
+            assertEq(params.guildToken, address(guild));
+            assertEq(params.auctionHouse, addresses.mainnet("AUCTION_HOUSE"));
+            assertEq(params.creditMinter, address(rateLimitedCreditMinter));
+            assertEq(params.creditToken, address(credit));
+        }
+    }
+
+    function testVoteForSDAIGauge() public {
+        uint256 mintAmount = governor.quorum(0);
+        /// setup
+        vm.prank(addresses.mainnet("TEAM_MULTISIG"));
+        rateLimitedGuildMinter.mint(address(this), mintAmount);
+        guild.delegate(address(this));
+
+        assertTrue(guild.isGauge(address(term)));
+        assertEq(guild.numGauges(), 1);
+        assertEq(guild.numLiveGauges(), 1);
+    }
+
+    function testAllocateGaugeToSDAI() public {
+        testVoteForSDAIGauge();
+
+        guild.incrementGauge(address(term), guild.balanceOf(address(this)));
+
+        assertEq(guild.totalWeight(), guild.balanceOf(address(this)));
+        assertTrue(guild.isUserGauge(address(this), address(term)));
+    }
+
+    function testSupplyCollateralUserOne(
+        uint128 supplyAmount
+    ) public returns (bytes32 loanId, uint128 suppliedAmount) {
+        supplyAmount = uint128(
+            _bound(
+                uint256(supplyAmount),
+                term.MIN_BORROW(),
+                rateLimitedCreditMinter.buffer()
```

Comment:
This upper bound will revert after the protocol is live, if there are multiple gauges, the debt ceiling of terms will be their relative gauge weight, and we won't be able to mint the whole rlcm buffer